packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1021,12 +1021,32 @@ export function attach(
       pendingSimulatedUnmountedIDs.length === 0 &&
       pendingUnmountedRootID === null
     ) {
-      // If we're currently profiling, send an "operations" method even if there are no mutations to the tree.
-      // The frontend needs this no-op info to know how to reconstruct the tree for each commit,
-      // even if a particular commit didn't change the shape of the tree.
+      // If we aren't profiling, we can just bail out here.
+      // No use sending an empty update over the bridge.
       if (!isProfiling) {
         return;
       }
+
+      const current = root.current;
+      const alternate = current.alternate;
+
+      // Certain types of updates bail out at the root without doing any actual render work.
+      // React should probably not call the DevTools commit hook in this case,
+      // but if it does- we can detect it and filter them out from the profiler.
+      // NOTE: Keep this logic in sync with the one in handleCommitFiberRoot()
+      const didBailoutAtRoot =
+        alternate !== null &&
+        alternate.expirationTime === 0 &&
+        alternate.childExpirationTime === 0;
```

Comment:
You already know this but this is very fragile and adds a dependency on fields we didn’t use before. I believe these might even be the fields Andrew is currently refactoring.

Can we make a proper fix in React itself instead?

Reply:
Actually never mind, I see we already depend on these fields below. 😳

Reply:
Oops, I literally *just* merged before your comment came in.

The reason I added this check in PR #17253 was because it could *also* improve the Profiler problem for older versions of React. If we think it's becoming unstable due to @acdlite refactoring, we should remove it and fix it properly in React: see PR #17267